### DISCUSSION TITLE
Stored changes not considered when querying within the same session

### DISCUSSION THREAD

**Discussion by ATDOR-GAT** [JasperFx/marten#2940] [Q&A]
Hi,

In our software we have a logic that opens a session, makes some changes to it and then runs additional queries that also need to consider the data that we have stored. In our LINQ-style queries the stored changes do not show up unless we have already committed the transaction. Unfortunately, we wanted to decide whether to commit based on the result of such queries.

Is there a way to make these changes "visible" for the query or do we have to redesign our transactions?

Reproduction: 
https://gist.github.com/ATDOR-GAT/38733d83e1a49f1397572fd218ff16e1

**Comment by jeremydmiller** [comment]
@ATDOR-GAT See this: https://martendb.io/documents/sessions.html#identity-map-mechanics

**Reply by ATDOR-GAT** [reply]
Hi Jeremy,

Thanks for the link. We have looked at the documentation before creating the reproduction (and this ticket) and particularly the section that you linked, and now I've read it through carefully, but I couldn't find a point that clarifies where we are wrong or have unreasonable expectations.

As in the reproduction, we use the `store.IdentitySession()` method, to open the session, so the identity map functionality should work. Let me summarize what we have found:
We open a session, load a document, modify its value and store it, but do not save the changes (commit) yet. Within the same session
✔ when we load the document again, and check the field, we see the change
✔ when we query the documents, filtering by the ID, we get back one result that has the updated value
✖ when we query the documents, filtering by the modified value, we get an empty collection. We assume that the query seems to run against the database, which doesn't see the change yet. The "staged" changes in the session are not considered.

Is this a limitation of the technology? Is there some option in the session we can configure so that LINQ queries can "see" the changes we already did in the session?

Thanks,
